snuba/web/db_query.py

```diff
@@ -193,10 +181,11 @@ def execute_query_with_deduplication(
         query_id = md5(force_bytes(query.format_sql())).hexdigest()
         with state.deduper(query_id) as is_dupe:
             timer.mark("dedupe_wait")
-            stats.update({"is_duplicate": is_dupe, "query_id": query_id})
-            return execute(query_id=query_id)
+            stats.update({"is_duplicate": is_dupe})
+            query_settings["query_id"] = query_id
```

Comment:
`query_id` doesn't need to be reported in stats as it's included via query settings here now: https://github.com/getsentry/snuba/blob/4796efbfb5bf6685b90b5a1480322676371bd712/snuba/web/db_query.py#L57